src/EnergyPlus/General.cc

```diff
@@ -1574,86 +1631,67 @@ void ScanForReports(EnergyPlusData &state,
 
             state.dataGeneral->EMSoutput = true;
 
-            {
-                auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(1));
-
-                if (SELECT_CASE_var == "NONE") {
-                    state.dataRuntimeLang->OutputEMSActuatorAvailSmall = false;
-                    state.dataRuntimeLang->OutputEMSActuatorAvailFull = false;
-                } else if (SELECT_CASE_var == "NOTBYUNIQUEKEYNAMES") {
-                    state.dataRuntimeLang->OutputEMSActuatorAvailSmall = true;
-                    state.dataRuntimeLang->OutputEMSActuatorAvailFull = false;
-                } else if (SELECT_CASE_var == "VERBOSE") {
-                    state.dataRuntimeLang->OutputEMSActuatorAvailSmall = false;
-                    state.dataRuntimeLang->OutputEMSActuatorAvailFull = true;
-
-                } else if (SELECT_CASE_var.empty()) {
-                    ShowWarningError(state, cCurrentModuleObject + ": Blank " + state.dataIPShortCut->cAlphaFieldNames(1) + " supplied.");
-                    ShowContinueError(state, R"( Legal values are: "None", "NotByUniqueKeyNames", "Verbose". "None" will be used.)");
-                    state.dataRuntimeLang->OutputEMSActuatorAvailSmall = false;
-                    state.dataRuntimeLang->OutputEMSActuatorAvailFull = false;
-                } else {
-                    ShowWarningError(state,
-                                     cCurrentModuleObject + ": Invalid " + state.dataIPShortCut->cAlphaFieldNames(1) + "=\"" +
-                                         state.dataIPShortCut->cAlphaArgs(1) + "\" supplied.");
-                    ShowContinueError(state, R"( Legal values are: "None", "NotByUniqueKeyNames", "Verbose". "None" will be used.)");
-                    state.dataRuntimeLang->OutputEMSActuatorAvailSmall = false;
-                    state.dataRuntimeLang->OutputEMSActuatorAvailFull = false;
-                }
+            AvailRpt CheckAvailRpt =
+                static_cast<AvailRpt>(getEnumerationValue(AvailRptNamesUC, UtilityRoutines::MakeUPPERCase(state.dataIPShortCut->cAlphaArgs(1))));
+            switch (CheckAvailRpt) {
```

Comment:
It's generally better not to use control flow to set flags. Just set the flags to the control flow condition. For example:

```cpp
OutputEMSActuatorAvailSmall = (CheckAvailRpt == AvailRpt::NotByUniqueNames);
OutputEMSActuatorAvailFull = (CheckAvailRpt == AvailRpt::Verbose);
```